src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { getFilteredPeople, getPersonParents } from '../utils/helpers';
+import { PeopleFilters } from '../components/PeopleFilters';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  const [serachParams, setSearchParams] = useSearchParams();
+  const query = serachParams.get('query') || '';
+  const sex = serachParams.get('sex') || '';
+  const centuries = serachParams.getAll('centuries') || [];
+
+  const loadPeopleFromServer = async () => {
+    try {
+      const peopleFromServer = await getPeople();
+      const peopleWithParents: Person[] = peopleFromServer.map(
+        person => getPersonParents(person, peopleFromServer),
+      );
+
+      setPeople(peopleWithParents);
+    } catch {
+      setHasError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
```

Comment:
It's better to move the creation of this function inside the `useEffect` where you use it.